src/plugins/layout/stateprovider.ts

```diff
@@ -129,3 +124,34 @@ export function createGenericLayoutState(node: Node, tag: string): ILayoutState
   };
   return layoutState;
 }
+
+export function shouldIgnoreNode(node: Node, shadowDom: ShadowDom): boolean {
+  let shadowNode = shadowDom.getShadowNode(getNodeIndex(node));
+  let ignore = false;
+  switch (node.nodeType) {
+    case Node.ELEMENT_NODE:
+      let tagName = (node as Element).tagName;
+      if (tagName === "SCRIPT" || tagName === "META") {
+        ignore = true;
+      }
+      break;
+    case Node.COMMENT_NODE:
+      ignore = true;
+      break;
+    default:
+      break;
+  }
+
+  // Ignore subtrees of ignored nodes (e.g. text with a <script> parent)
+  if (!ignore) {
+    let parentIndex = getNodeIndex(node.parentNode);
+    if (parentIndex !== null) {
+      let parentShadowNode = shadowDom.getShadowNode(parentIndex);
+      assert(!!parentShadowNode, "shouldIgnoreNode", "parentShadowNode is missing");
+      if (parentShadowNode && parentShadowNode.ignore && parentShadowNode.node !== document) {
+        ignore = true;
```

Comment:
There is no loop to break

---
In reply to: [131241578](https://github.com/Microsoft/clarity-js/pull/25#discussion_r131241578) [](ancestors = 131241578)